bioinf_mega_tools.py

```diff
@@ -75,14 +112,20 @@ def filter_fastq(
     for filter in the scale phred33,
     default = 0
     """
-    filtered_seqs = dict()
-    seqs = scripts.rw_fastq.read_fastq(input_fastq)
-
-    for seq_id, (seq, qual) in seqs.items():
-        if (
-            is_filter_gc(seq, gc_bounds)
-            and is_filter_length(seq, length_bounds)
-            and is_filter_quality(qual, quality_threshold)
-        ):
-            filtered_seqs[seq_id] = (seq, qual)
-    scripts.rw_fastq.write_fastq(output_fastq, filtered_seqs)
+    if not os.path.exists("./filtered/"):
+        os.mkdir("./filtered/")
+    if not isinstance(gc_bounds, tuple):
+        gc_bounds = (0, gc_bounds)
+    if not isinstance(length_bounds, tuple):
+        length_bounds = (0, length_bounds)
+
+    with open(input_fastq) as input_handle:
+        with open(os.path.join("./filtered/", output_fastq), "w") as output_handle:
+            for record in SeqIO.parse(input_handle, "fastq"):
+                qualities = record.letter_annotations["phred_quality"]
+                if (
+                    gc_bounds[0] <= GC(record.seq) <= gc_bounds[1]
```

Comment:
`GC` устаревший, поэтому лучше так
```suggestion
                    gc_bounds[0] <= gc_fraction(record.seq) * 100 <= gc_bounds[1]
```
(только импортировав, конечно)
`* 100` потому, что возвращает долю (0, 1)